docs/proposals/28-autonomous-shoot-clusters.md

```diff
@@ -0,0 +1,495 @@
+> [!NOTE]
+> `kubeadm` goes one step further and implements a discovery process for the certificate authority of the control plane.
+> For this purpose, so-called "discovery tokens" are used.
+>
+> First, the CA certificate is retrieved via an insecure connection from a `ConfigMap` that got created as part of the `kubeadm init` process.
+> Next, the downloaded CA certificate is validated based on a hash of the public key that must be provided as part of `kubeadm join`.
+> Finally, as additional validation, the CA certificate is retrieved via secure connection and compared with the CA retrieved initially.
+>
+> We might implement a similar discovery mechanism for `gardenadm` as this project evolves.
+> However, we plan to first focus on the other, "more functional" aspects.
+
+#### `gardenadm join`
+
+Joining additional worker nodes is very similar to how it works for regular shoot clusters.
+This command instruments `gardener-node-agent` reconciliation logic to write the `systemd` units and needed files for the [`gardener-node-init`](../concepts/node-agent.md#installation-and-bootstrapping) component onto the disk of the machine.
+When this unit is started, it extracts the `gardener-node-agent` binary from its container image, and creates the `systemd` unit for it.
+`gardener-node-agent` will then use the provided bootstrap token to fetch the `OperatingSystemConfig` from a `Secret` residing in the `kube-system` namespace of the cluster.
+This `OperatingSystemConfig` contains the actual components like `kubelet`, `valitail`, etc.
+With the help of the same bootstrap token, `kubelet` will come up and register its `Node` resource, hence, successfully joins the cluster.
+
+The process for joining additional control plane nodes is very similar.
+The main difference is that the `OperatingSystemConfig` will contain the manifests of the static pods for the minimal standard Kubernetes control plane.
+Depending on the `etcd` management, it may be necessary to have individual `OperatingSystemConfig`s for each control plane node.
+However, this is not yet fully clear and needs further investigation (see [`etcd` Management via `etcd-druid`](#etcd-management-via-etcd-druid)).
+Apart from this, `etcd` also needs to be re-configured to have the new node join the `etcd` cluster.
+
+#### `gardenadm connect`
+
+This command deploys a `gardenlet` into the autonomous shoot cluster and connects it to an existing garden cluster.
+As part of this operation, the corresponding `Shoot` resource will be created in the garden cluster.
+Afterwards, the `gardenlet` will reconcile the autonomous shoot cluster similarly to regular shoot clusters.
+
+> [!NOTE]
+> When there is no existing Gardener system yet (for example, because the created autonomous shoot cluster is the initial cluster), then [`gardener-operator`](../concepts/operator.md) must be deployed first, and the [`Garden`](../../example/operator/20-garden.yaml) must be created, before the `gardenlet` can be deployed.
```

Comment:
Will the initial cluster's gardenlet register itself as a Shoot with a corresponding shoot resource in its own virtual cluster? Does `k delete shoot initial-cluster` make sense and if yes under what circumstances?

Reply:
We assume that `gardenlet` will be connected to a garden and registers itself as the "seed" for the autonomous shoot cluster. Whether the garden is running on the autonomous shoot cluster itself or somewhere else is not that important from my point of view.
As we currently do not aim to support a fully automated autonomous shoot cluster creation (see https://github.com/gardener/gardener/blob/68ccf134a903568bb724f0926d5c3dfa2e470379/docs/proposals/28-autonomous-shoot-clusters.md#non-goals), I think it makes sense that deletion of autonomous shoot clusters may also require some manual steps. In [medium touch scenarios](https://github.com/gardener/gardener/blob/68ccf134a903568bb724f0926d5c3dfa2e470379/docs/proposals/28-autonomous-shoot-clusters.md#medium-touch), it might still be possible to perform some sensible steps to delete the nodes and infrastructure.